airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -144,6 +138,22 @@ public OUTPUT get() {
     }
   }
 
+  private void saveWorkflowIdForCancellation() throws IOException {
+    // If the jobId is not a number, it means the job is a synchronous job. No attempt is created for
+    // it, and it cannot be cancelled, so do not save the workflowId. See SynchronosSchedulerClient.java
+    // for info.
+    if (NumberUtils.isCreatable(jobRunConfig.getJobId())) {
+      final Database jobDatabase = new JobsDatabaseInstance(
```

Comment:
it feels like kinda big deal that we are accessing the jobs db from within temporal. not necessarily bad, but it is different from how we've been thinking about it. just want to make sure we do it consciously.

Reply:
You are right - it's a big change. See my above comment about accessing the workflow id from outside temporal/before the workflow is created.